Indicators/ZigZag.cs

```diff
@@ -0,0 +1,175 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// The ZigZag indicator identifies significant turning points in price movements,
+    /// filtering out noise using a sensitivity threshold and a minimum trend length.
+    /// It alternates between high and low pivots to determine market trends.
+    /// </summary>
+    public class ZigZag : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        /// <summary>
+        /// Stores the most recent high pivot value in the ZigZag calculation.
+        /// Updated whenever a valid high pivot is identified.
+        /// </summary>
+        public decimal HighPivot { get; set; }
```

Comment:
Just a thought but would the user like to know which was the last/current pivot? Either a high or a low?
Perhaps we can have a data class to hold the pivot value (a decimal) and maybe an enum for `High`/`Low`?

Reply:
I need to expose these values to update them in case a lower or higher point is found compared to the current one. This happens because we process one value at a time, so these variables store the latest updated low/high pivots. Now, if the user only needs the last pivot, they can simply use .Current.Value from the indicator.